src/interop/js-interop/js-types.md

```diff
@@ -0,0 +1,189 @@
+---
+title: JS Types
+description: Usage information about the core types in JS interop.
+---
+
+## Motivation
+
+Dart values and JS values belong to separate domains. When compiling to Wasm,
+they belong in separate *runtimes* as well. As such, users should treat JS
+values as foreign types. In order to provide types for JS values, we use the set
+of JS types available in `dart:js_interop`. These are extension types prefixed
+with `JS` and are called "JS types".
+
+Importantly, these types have compiler-specific representation types. This means
+that their runtime type will differ based on whether code is compiled to Wasm or
+JS. In order to interact with and examine these JS values, we use `external`
+interop members.
+
+## Type Hierarchy
+
+JS types form a natural type hierarchy:
+
```

Comment:
I like it! If we're not going to link stuff directly, it might be better to make this a diagram, but I agree the search benefits are useful. I might come back to this - do we have goto program to create such a diagram?

Reply:
I don't think we have an official go-to, but some recommendations from the team are:

- make it in Slides and do a screenshot
- Google Draw
- (highly recommended) Figma with the FigJam feature (less complicated than just Figma)
  - in this case you also need to provide the source for that diagram in the [diagrams folder](https://github.com/dart-lang/site-www/tree/main/diagrams)) using Omnigraffle (instructions are in the readme there). 
  - disclaimer: I've never done this myself

Reply:
As a note, any format is probably to put in the diagrams folder as long as it is loadable again and there is some breadcrumb to track what app to open it in (extension or commit message is fine).

I've also used https://app.diagrams.net/ for an experience like Google Draw but a bit more feature rich.

FigJam sounds interesting! I'll have to check it out :o